File: apple/Sources/ReactNativeWatchOS/Modifiers/TabViewStyle.swift
import SwiftUI

/// SwiftUI `.tabViewStyle(_:)`. Maps the `style` string to a tab-view
/// style. `.page` / `.carousel` are watchOS 7+ and `.verticalPage` is
/// watchOS 9+ — all available on the watchOS 9 deployment target, so no
/// availability gating is required. Unknown values fall back to
/// `.automatic`.
enum RNWTabViewStyleModifier {
    @MainActor
    static func register(into r: RNWModifierRegistry) {
        r.register("tabViewStyle") { view, params, _ in
            switch params.string("style") {
            case "page":
                return AnyView(view.tabViewStyle(.page))
            case "verticalPage":
                // `.verticalPage` is watchOS 10+; fall back to `.page` below.
                if #available(watchOS 10.0, *) {
                    return AnyView(view.tabViewStyle(.verticalPage))
                }
                return AnyView(view.tabViewStyle(.page))
            case "carousel":
                return AnyView(view.tabViewStyle(.carousel))
            default:
                return AnyView(view.tabViewStyle(.automatic))
            }
        }
    }
}

File: src/modifiers/tabViewStyle.ts
import { createModifier } from './createModifier';

export type TabViewStyle =
  | 'automatic'
  | 'page'
  | 'verticalPage'
  | 'carousel';

/// SwiftUI `.tabViewStyle(_:)`. Sets the visual style applied to `TabView`s
/// within the view. `'automatic'` defers to the platform default;
/// `'verticalPage'` and `'carousel'` are watchOS-specific styles.
export function tabViewStyle(style: TabViewStyle = 'automatic') {
  return createModifier('tabViewStyle', { style });
}
